kernel/packages/shared/comms/index.ts

```diff
@@ -691,6 +703,31 @@ export async function connect(userId: string) {
   }
 }
 
+function realmString(realm: Realm) {
+  return realm.catalystName + '-' + realm.layer
+}
+
+function handleReconnectionError() {
+  const store: Store<RootState> = window.globalStore
+  const realm = getRealm(store.getState())
+
+  if (realm) {
+    store.dispatch(markCatalystRealmConnectionError(realm))
+  }
+
+  const candidates = getAllCatalystCandidates(store.getState())
+
+  const otherRealm = pickCatalystRealm(candidates)
```

Comment:
Should we update `pickCatalystRealm` to consider the candidate status?

Reply:
Done!